apps/server/src/shared/infra/identity-management/keycloak-configuration/service/keycloak-migration.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@nestjs/common';
+import { AccountService } from '@src/modules/account/services/account.service';
+import { AccountDto } from '@src/modules/account/services/dto';
+
+@Injectable()
+export class KeycloakMigrationService {
+	constructor(private readonly accountService: AccountService) {}
+
+	async migrate(): Promise<[number, string[]]> {
+		const amount = 100;
+		let skip = 0;
+		let foundAccounts = 1;
+		let migratedAccounts = 0;
+		let accounts: AccountDto[] = [];
+		const errorList: string[] = [];
+		while (foundAccounts > 0) {
+			// eslint-disable-next-line no-await-in-loop
+			({ accounts, total: foundAccounts } = await this.accountService.searchByUsernamePartialMatch('', skip, amount));
```

Comment:
is there some recovery?

e.g. aborting process and resume at the same account and skip already migrated ones?

Reply:
Nope, not implemented. The idea is to keep the script as simple as possible. It would simply iterate all accounts (alphabetical order). What can quickly be added is to skip the first n accounts and to specify a search pattern, what should come in handy.